File: Argus/Spring-Backend/src/main/java/com/springboot/SpringBackend/service/ImageService.java
package com.springboot.SpringBackend.service;

import com.springboot.SpringBackend.model.Image;

import java.util.List;
import java.util.Optional;

public interface ImageService {

    List<Image> listAllImages();

    Optional<Image> getImageById(Long id);

    Image createImage(Image x);

    Image updateImage(Image x);

    void deleteImage(Image x);

    void deleteImageById(Long id);
}

File: Argus/Angular-Frontend/src/app/model/image.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, throwError } from 'rxjs';
@Injectable({
  providedIn: 'root'
})
export class ImageService {
  private baseUrl = 'http://localhost:9000/springboot/api/images';

  constructor(private http: HttpClient) { }

  getImageById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  addImage(image: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, image);
  }

  updateImage(id: number, value: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }

  deleteImage(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  getImageList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }
}
